roles/v2/pool/src/main.rs

```diff
@@ -135,19 +137,23 @@ async fn main() {
         }
     };
 
+    let (status_tx, status_rx) = unbounded();
     let (s_new_t, r_new_t) = bounded(10);
     let (s_prev_hash, r_prev_hash) = bounded(10);
     let (s_solution, r_solution) = bounded(10);
     let (s_message_recv_signal, r_message_recv_signal) = bounded(10);
     info!("Pool INITIALIZING with config: {:?}", &args.config_path);
+
     TemplateRx::connect(
         config.tp_address.parse().unwrap(),
         s_new_t,
         s_prev_hash,
         r_solution,
         r_message_recv_signal,
+        status::Sender::Upstream(status_tx.clone()),
     )
-    .await;
+    .await
+    .unwrap();
```

Comment:
Can you add a comment here along the lines of "unwrap() ok here because we want the pool to die if it can't connect to the TP"?

Reply:
> Can you add a comment here along the lines of "unwrap() ok here because we want the pool to die if it can't connect to the TP"?

i don't think pool should die even if its dead in one sense, should keep trying connect to TP in list
but also need some way signal Ops is needed . trigger a `alert_TP_down.sh` that ops then can choose use text mail phone etc. for instant notification

Reply:
In general I agree. I added this gh issue that should resolve this - https://github.com/stratum-mining/stratum/issues/427 This PR isn't about fixing all of the issues so I'm ok with letting this one get merged as long as we have a gh issue documenting the problem and we can fix it later.

Reply:
> In general I agree. I added this gh issue that should resolve this - #427 This PR isn't about fixing all of the issues so I'm ok with letting this one get merged as long as we have a gh issue documenting the problem and we can fix it later.

should We introduce a backlog.md for will make complete later TODO 's

Reply:
> > In general I agree. I added this gh issue that should resolve this - #427 This PR isn't about fixing all of the issues so I'm ok with letting this one get merged as long as we have a gh issue documenting the problem and we can fix it later.
> 
> should We introduce a backlog.md for will make complete later TODO 's

or just in code have the TODO 's

becouse this GH issues and PR's is messy . ppl close issues to reopen in own name and reference issues to new issues that gets to be spagetti in end ...

a issue should be kept as a issue and further the issue in the issue . agree ?

Reply:
I don't entirely understand. I think that if GH issues and PRs are messy then we should get better and organizing them. I wouldn't want to have to maintain a backlog.md in addition to GH issues.